build.sbt

```diff
@@ -272,176 +250,139 @@ lazy val examplesJVM = examples.jvm
-  .settings(dottySettings)
+  .settings(stdSettings(name = "zio-config-typesafe", enableCrossProject = true))
+  .settings(enableZIO())
+  .settings(scala3Settings)
   .settings(
-    crossScalaVersions --= Seq("2.11"),
-    libraryDependencies ++= Seq(
-      "com.typesafe" % "config"       % "1.4.2",
-      "dev.zio"     %% "zio-test"     % zioVersion % Test,
-      "dev.zio"     %% "zio-test-sbt" % zioVersion % Test
-    ),
-    testFrameworks := Seq(new TestFramework("zio.test.sbt.ZTestFramework"))
+    libraryDependencies ++= Seq("com.typesafe" % "config" % "1.4.2")
   )
   .dependsOn(zioConfig % "compile->compile;test->test")
 
 lazy val zioConfigTypesafeJVM = zioConfigTypesafe.jvm
-  .settings(dottySettings)
+  .settings(scala3Settings)
 
-lazy val zioConfigYaml    = crossProject(JVMPlatform)
+lazy val zioConfigYaml = crossProject(JVMPlatform)
   .in(file("yaml"))
-  .settings(stdSettings("zio-config-yaml"))
-  .settings(crossProjectSettings)
+  .settings(stdSettings(name = "zio-config-yaml", enableCrossProject = true))
+  .settings(enableZIO())
   .settings(
-    crossScalaVersions --= Seq("2.11"),
     libraryDependencies ++= Seq(
-      "org.snakeyaml" % "snakeyaml-engine" % "2.6",
-      "dev.zio"      %% "zio-test"         % zioVersion % Test,
-      "dev.zio"      %% "zio-test-sbt"     % zioVersion % Test
-    ),
-    testFrameworks := Seq(new TestFramework("zio.test.sbt.ZTestFramework"))
+      "org.snakeyaml" % "snakeyaml-engine" % "2.6"
+    )
   )
   .dependsOn(zioConfig % "compile->compile;test->test")
 
-lazy val zioConfigYamlJVM = zioConfigYaml.jvm
-  .settings(dottySettings)
+lazy val zioConfigYamlJVM = zioConfigYaml.jvm.settings(scala3Settings)
 
-lazy val zioConfigXml    = crossProject(JVMPlatform)
+lazy val zioConfigXml = crossProject(JVMPlatform)
   .in(file("xml"))
-  .settings(stdSettings("zio-config-xml"))
-  .settings(crossProjectSettings)
+  .settings(stdSettings(name = "zio-config-xml", enableCrossProject = true))
+  .settings(enableZIO())
   .settings(
-    crossScalaVersions --= Seq("2.11"),
-    libraryDependencies ++= Seq(
-      "dev.zio" %% "zio-parser"   % "0.1.8",
-      "dev.zio" %% "zio-test"     % zioVersion % Test,
-      "dev.zio" %% "zio-test-sbt" % zioVersion % Test
-    ),
-    testFrameworks := Seq(new TestFramework("zio.test.sbt.ZTestFramework"))
+    libraryDependencies ++= Seq("dev.zio" %% "zio-parser" % "0.1.8")
   )
   .dependsOn(zioConfig % "compile->compile;test->test")
 
-lazy val zioConfigXmlJVM = zioConfigXml.jvm
-  .settings(dottySettings)
+lazy val zioConfigXmlJVM = zioConfigXml.jvm.settings(scala3Settings)
 
-lazy val zioConfigScalaz    = crossProject(JSPlatform, JVMPlatform, NativePlatform)
+lazy val zioConfigScalaz = crossProject(JSPlatform, JVMPlatform, NativePlatform)
   .in(file("scalaz"))
-  .settings(stdSettings("zio-config-scalaz"))
-  .settings(crossProjectSettings)
+  .settings(stdSettings(name = "zio-config-scalaz", enableCrossProject = true))
+  .settings(enableZIO())
```

Comment:
I think its mostly to help defining modules easily without relying on adhoc sbt settings. As mentioned above, as a maintainer I am far relieved if I can just simply define `module(...)` using the plugin, or if needed `jvmModule`, `jsModule` etc.